### ISSUE TITLE
Suppress `PossiblyUnusedMethod` for #[Scope]-attributed methods

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#874]
## Problem

A model query scope declared with the modern `#[Scope]` attribute is reported as `PossiblyUnusedMethod` (and `UnusedMethod` under `findUnusedCode=true`):

```
ERROR: PossiblyUnusedMethod
Cannot find any calls to method App\Models\Post::published (see https://psalm.dev/087)
    protected function published(Builder $query): void
```

Reproducer:

```php
use Illuminate\Database\Eloquent\Attributes\Scope;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;

final class Post extends Model
{
    /** @param Builder<self> \$query */
    #[Scope]
    protected function published(Builder \$query): void
    {
        \$query->where('published_at', '<=', now());
    }
}

// Call site Psalm doesn't observe as a reference back to Post::published():
Post::query()->published()->get();
```

Eloquent dispatches scopes through `Builder::callNamedScope()` / `Builder::__call()` via reflection and `Model::__call()`, so Psalm cannot link the `->published()` call site to the model method. The plugin already understands `#[Scope]` for type/visibility purposes (see `BuilderScopeHandler`, `ModelMethodHandler`, `CustomBuilderMethodHandler`), but does not suppress the unused-method finding.

## Proposed fix

Extend `SuppressHandler` (`src/Handlers/SuppressHandler.php`) with a new pass over Eloquent model classes (analogous to `suppressEloquentAccessorMethods()`) that suppresses `PossiblyUnusedMethod` (and `UnusedMethod`) on any method whose `MethodStorage` carries the `Illuminate\Database\Eloquent\Attributes\Scope` attribute.

Visibility gating: route through `suppressInternalDispatchMethod()` (allow public + protected, keep private flagged). At runtime Eloquent invokes the scope on the model instance, so private overrides would fatal anyway — no reason to silence them.

## Out of scope (follow-up)

- Legacy `scopeXxx()` methods. Same root cause, same fix shape, but worth a separate issue/PR so the diff and the test additions stay focused.
- Methods that participate in scopes only by name on a relation (e.g. `\$relation->published()` where the related model has the scope); the existing handlers cover the type side, the suppression side mirrors the current issue.

## Acceptance

- New PHPT type test under `tests/Type/tests/` covering a `#[Scope]`-attributed method declared `public` and `protected`, asserting no `PossiblyUnusedMethod` is reported.
- Negative case: a `private` method with `#[Scope]` still reports (it's a real bug at runtime).
- No regression in existing scope handler tests.

**Comment by alies-dev** [comment]
Subset of #865 (PossiblyUnusedMethod suppression umbrella for marker-less / reflection-dispatched Laravel patterns). Tracking #865 as the umbrella; #869 is the peer that locks in the suppressions with a `findUnusedCode=true` test config so this can't silently regress.
